src/video/vid_riva128.c

```diff
@@ -0,0 +1,1072 @@
+
+    struct
+    {
+        int sda, scl;
+    } i2c;
+
+    struct
+    {
+        uint8_t rma_access_reg[4];
+        uint8_t rma_mode;
+        uint32_t rma_dst_addr;
+        uint32_t rma_data;
+    } rma;
+
+	struct 
+	{
+		uint32_t intr;
+		uint32_t intr_en;
+		uint32_t intr_line;
+		uint32_t enable;
+	} pmc;
+
+	struct
+	{
+		uint32_t intr, intr_en;
+
+		uint64_t time;
+		uint32_t alarm;
+
+		uint16_t clock_mul, clock_div;
+	} ptimer;
+
+	struct
+	{
+		uint32_t nvpll, mpll, vpll;
+	} pramdac;
+
+	pc_timer_t mtimer;
+
+	uint64_t mtime;
+} riva128_t;
+
+static uint8_t riva128_in(uint16_t addr, void *p);
+static void riva128_out(uint16_t addr, uint8_t val, void *p);
+
+
+static uint8_t 
+riva128_pci_read(int func, int addr, void *p)
+{
+    riva128_t *riva128 = (riva128_t *)p;
+    // svga_t *svga = &riva128->svga;
+
+    switch (addr) {
+	case 0x00: return 0xd2; /*nVidia*/
+	case 0x01: return 0x12;
+
+	case 0x02: return 0x18;
+	case 0x03: return 0x00;
+	
+	case 0x04: return riva128->pci_regs[0x04] & 0x37; /*Respond to IO and memory accesses*/
+	case 0x05: return riva128->pci_regs[0x05] & 0x01;
+
+	case 0x06: return 0x20;
+	case 0x07: return 0x02; /*Fast DEVSEL timing*/
+
+	case 0x08: return 0x00; /*Revision ID*/
+	case 0x09: return 0x00; /*Programming interface*/
+
+	case 0x0a: return 0x00; /*Supports VGA interface*/
+	case 0x0b: return 0x03;
+
+	case 0x13: return riva128->mmio_base >> 24;
+
+	case 0x17: return riva128->lfb_base >> 24;
+
+	case 0x2c: case 0x2d: case 0x2e: case 0x2f:
+		return riva128->pci_regs[addr];
+
+	case 0x30: return (riva128->pci_regs[0x30] & 0x01); /*BIOS ROM address*/
+	case 0x31: return 0x00;
+	case 0x32: return riva128->pci_regs[0x32];
+	case 0x33: return riva128->pci_regs[0x33];
+
+	case 0x3c: return riva128->int_line;
+	case 0x3d: return PCI_INTA;
+
+	case 0x3e: return 0x03;
+	case 0x3f: return 0x01;
+    }
+
+    return 0x00;
+}
+
+
+static void 
+riva128_recalc_mapping(riva128_t *riva128)
+{
+    svga_t *svga = &riva128->svga;
+        
+    if (!(riva128->pci_regs[PCI_REG_COMMAND] & PCI_COMMAND_MEM)) {
+	pclog("PCI mem off\n");
+        mem_mapping_disable(&svga->mapping);
+        mem_mapping_disable(&riva128->mmio_mapping);
+        mem_mapping_disable(&riva128->ramin_mapping);
+        mem_mapping_disable(&riva128->linear_mapping);
+	return;
+    }
+
+    pclog("PCI mem on\n");
+    pclog("riva128->mmio_base = %08X\n", riva128->mmio_base);
+    if (riva128->mmio_base)
+	mem_mapping_set_addr(&riva128->mmio_mapping, riva128->mmio_base, 0x1000000);
+    else
+        mem_mapping_disable(&riva128->mmio_mapping);
+
+    pclog("riva128->lfb_base = %08X\n", riva128->lfb_base);
+    if (riva128->lfb_base) {
+	mem_mapping_set_addr(&riva128->linear_mapping, riva128->lfb_base, 0x0c00000);
+	mem_mapping_set_addr(&riva128->ramin_mapping, riva128->lfb_base + 0x0c00000, 0x0400000);
+    } else {
+        mem_mapping_disable(&riva128->linear_mapping);
+        mem_mapping_disable(&riva128->ramin_mapping);
+    }

```

Comment:
check indenting

Reply:
The indenting looks fine to me. :/

Reply:
hm. for me it looks like
```
if (riva128->lfb_base) {
mem_mapping_set_addr(&riva128->linear_mapping, riva128->lfb_base, 0x0c00000);
} else {
  mem_mapping_disable(&riva128->linear_mapping);
}
```
i.e. the "else" is indented by 4 spaces, the "if" part not at all. (or rather 4 spaces vs. 8 spaces, if you look at the total indentation)